src/core/utils/binary-writer-utils.ts

```
@@ -143,4 +143,31 @@ export class BinaryWriter {
+        .map(
+          (b, j) =>
+            b.toString(16).padStart(2, "0") + ((j + 1) % 8 === 0 ? "  " : " ")
+        )
+        .join("")
+        .padEnd(bytesPerLine * 3 + Math.floor(bytesPerLine / 8), " ");
+
+      const ascii = chunk
+        .map((b) => (b >= 0x20 && b <= 0x7e ? String.fromCharCode(b) : "."))
+        .join("");
+
+      return [...out, `${i.toString(16).padStart(4, "0")}: ${hex} ${ascii}`];
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)